scripts/generator-adapter/generators/app/templates/src/endpoint/endpoint-router.ts

```diff
@@ -0,0 +1,33 @@
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { TransportRoutes } from '@chainlink/external-adapter-framework/transports'
+import { config } from '../config'
+import overrides from '../config/overrides.json'
+<% for(let i=0; i<inputTransports.length; i++) {%>
+import { <%= inputTransports[i].name %> } from '../transport/<%= inputEndpointName %>-<%= inputTransports[i].type %>' <% }
+%>
+
+<%- include ./base.ts %>
+<%/* [THIS WILL NOT BE RENDERED]. Indeed, below is a very unpleasant code, but the main purpose of it is to have everything nicely formatted, otherwise EJS puts unwanted new-lines for every 'if' placeholder */%>
```

Comment:
Not sure I follow 🤔

Reply:
That is a comment that will NOT be included in the generated adapter. The syntax bellow that comment is really weird and could be simplified but the way ejs works is that it will create placeholder empty spaces and new lines for conditional statements . By default `if` statement creates new line character, so the rendered code would like something like this

```typescript
export const endpoint = new CryptoPriceEndpoint({
  name: 'crypto',

  aliases: ['crypto-batched', 'batched', 'batch', 'price'],

  transport,

  inputParameters: cryptoInputParams,

  overrides: overrides.coingecko,

})
```

There is a special workaround to render everything normally and ignore new lines. The comment there just tells to be careful when changing or formatting that part, as it will most likely break the final formatting.  This was the first place with this syntax, however other templates also have it, so i guess it applies for everything and comment can be removed.